### DISCUSSION TITLE
For loops

### DISCUSSION THREAD

**Discussion by lazytanuki** [elkowar/eww#442] [Q&A] [answered]
Hi and thanks for your work on this !

I am trying to use for loops as mentioned in [this part of the documentation about adding dynamic content](https://elkowar.github.io/eww/configuration.html?highlight=json#generating-a-list-of-widgets-from-json-using-for). However, when using the example inside of a test widget, I get the following error:

```
error: No variable named `entry` in scope
   ┌─ .../eww/eww.yuck:18:10
   │
18 │     (for entry in foo
   │          ───── Used here
   │
   → Hint: Either define it as a global variable, or add it to the argument-list of your `defwidget` and pass it as an argument
   → Hint: If you meant to use the literal value "entry", surround the value in quotes
```

Here is the part where I am trying to use it:

```yuck
(defvar my-json "[1, 2, 3]")

(defwidget testwidget []
  (box
    (for entry in my-json
      (button :onclick "notify-send 'click' 'button ${entry}'"
        entry)))
  )
```

Did I miss something ?

Thanks !

**Comment by elkowar** [comment, marked as answer]
Are you on latest git Master? Loops where only added very recently

**Reply by lazytanuki** [reply]
Oddly enough, the binary provided by the AUR package I installed returns version 0.2.0 with the `-V` arg, but I had to clone and compile it myself for the feature to be here. Anyway, thanks, I'm already having a good time using eww !
